### ISSUE TITLE
Is Automatic Mask Generator "trainable"?

### ISSUE THREAD

**Issue by ckapelonis02** [facebookresearch/sam2#602]
Hello!
First of all just wanted to say great job by the developers!

I am trying to fine-tune SAM2 but I noticed that all the online tutorials use the image predictor method instead of the automatic mask generator as inference on the fine-tuned model.

I am somewhat confused as to whether fine-tuning SAM2 (with masks, points etc.) can produce something that can be used with the automatic mask generator. Or is it the case that you can only use the image predictor?

Thanks in advance!

**Comment by ckapelonis02** [comment]
### UPDATE:

So if I understand this correctly `SAM2AutomaticMaskGenerator` wraps around `SAM2ImagePredictor`, automating the process of prompting.

My solution was to tweak the automatic mask generator code so that it takes another argument which is **load_model** as such:

```python
class SAM2AutomaticMaskGenerator:
    def __init__(
        self,
        model: SAM2Base,
        points_per_side: Optional[int] = 32,
...
        multimask_output: bool = True,
        load_model: str = None
    ) -> None:
```

And inside the body of `__init__`:

```python
...
        self.predictor = SAM2ImagePredictor(
            model,
            max_hole_area=min_mask_region_area,
            max_sprinkle_area=min_mask_region_area,
        )
        if (load_model is not None):
            self.predictor.model.load_state_dict(torch.load(load_model))
        self.points_per_batch = points_per_batch
...
```

Then just invoke the generator like this:

```python
sam2_model = build_sam2(model_cfg, sam2_checkpoint, device="cuda")

mask_generator = SAM2AutomaticMaskGenerator(
    sam2_model,
    load_model="model.torch"
    )
```

Or specify `None` if you do not want to load your pretrained model.
